CatAnalyzer/python/analyzers/ntuple_cff.py

```diff
@@ -0,0 +1,161 @@
+import FWCore.ParameterSet.Config as cms
+
+def ntupler_load(process, name):
+    process.load("CATTools.CatAnalyzer.analyzers.ntuple_cfi")
+    for typeName in ['int', 'ints', 'float', 'floats']:
+        for key, val in getattr(process.ntuple, typeName).parameters_().iteritems():
+            setattr(getattr(process.ntuple, typeName), key, cms.InputTag(val.value().replace('eventSel', name)))
+    for key, val in process.ntuple.cands.parameters_().iteritems():
+        getattr(process.ntuple.cands, key).src = cms.InputTag(val.src.value().replace('eventSel', name))
+    return process
+
+def ntupler_addVarsTTLL(process, name):
```

Comment:
This part is to add dilepton kinematic reconstruction. this is turned on only for the dilepton channel. You can do the similar thing for the semileptonic channel if you have a similar EDProducer.